### ISSUE TITLE
not able lower tf.Conv2DBackpropInput with dynamic shapes

### ISSUE THREAD

**Issue by ArunaKote** [google/iree#7052]
model looks like:
model = tf.keras.Sequential([ 
     tf.keras.layers.TimeDistributed(tf.keras.layers.Conv2DTranspose(10, (2, 2), strides=2, activation="relu"),input_shape=(None, 5,5,3),name="mrcnn_mask_deconv")
])
<unknown>:0: error: The following Tensorflow operations still remain:
        tf.Conv2DBackpropInput (count: 1)

If the shape is static I able to lwero to HLO:
func @conv2d_backprop_input(
    %filter: tensor<3x3x1x32xf32>,
    %out_backprop: tensor<100x26x26x32xf32>
  ) -> tensor<100x28x28x1xf32> {
  %input_sizes = "tf.Const" () { value = dense<[100,28,28,1]> : tensor<4xi32> } : 
() -> tensor<4xi32>
  %result = "tf.Conv2DBackpropInput"(%input_sizes, %filter, %out_backprop) {
    data_format = "NHWC",
    dilations = [1, 1, 1, 1],
    explicit_paddings = [],
    padding = "VALID",
    strides = [1, 1, 1, 1],
    use_cudnn_on_gpu = true
  } : (tensor<4xi32>, tensor<3x3x1x32xf32>, tensor<100x26x26x32xf32>) -> 
tensor<100x28x28x1xf32>
  return %result : tensor<100x28x28x1xf32>
}

HLO:
builtin.module  {
  builtin.func @conv2d_backprop_input(%arg0: tensor<3x3x1x32xf32>, %arg1: tensor<100x26x26x32xf32>) -> tensor<100x28x28x1xf32> {
    %0 = mhlo.constant dense<[100, 28, 28, 1]> : tensor<4xi32>
    %1 = "mhlo.reverse"(%arg0) {dimensions = dense<[0, 1]> : tensor<2xi64>} : (tensor<3x3x1x32xf32>) -> tensor<3x3x1x32xf32>
    %2 = mhlo.convolution(%arg1, %1) dim_numbers = [b, 0, 1, f]x[0, 1, o, i]->[b, 0, 1, f], window = {stride = [1, 1], pad = [[2, 2], [2, 2]], lhs_dilate = [1, 1], rhs_dilate = [1, 1]} {batch_group_count = 1 : i64, feature_group_count = 1 : i64} : (tensor<100x26x26x32xf32>, tensor<3x3x1x32xf32>) -> tensor<100x28x28x1xf32>
    return %2 : tensor<100x28x28x1xf32>
  }
} 

I have attached log file and python file
[backprop.zip](https://github.com/google/iree/files/7169778/backprop.zip)
